code/common/pcapdecoder.q

```diff
@@ -0,0 +1,87 @@
+\d .pcap
+
+// size of headers in bytes and dict of protocol code conversions
+globheader: 24;
+packetheader: 16;
+allcodes:(enlist 6)!(enlist `TCP);
+
+
+buildtable:{[file]
+ // initial x and byte cut points removed from array list to make table
+ // gettablerow is iterated over each datapacket, extracting data
+
+ data:1_ last each {[n]
+  filebytesize: count n;
+  gettablerow[n;]\[{y>(first x[0])+40}[;filebytesize];(),0]    
+  } read1 file;
+ 
+ data: update sym:` from data;
+ `time`sym xcols data
+ }
+
+
+gettablerow:{[n;x]  // data for a single row
```

Comment:
Could you add a comment somewhere what `n` & `x` are? Or give them more descriptive names - most other variables seem in this script seem to have pretty good descriptive names